src/app/forms/dynamic-form/service/terra-form-field-control.service.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable } from '@angular/core';
+import { TerraFormFieldBase } from '../data/terra-form-field-base';
+import {
+    FormBuilder,
+    FormControl,
+    FormGroup,
+    ValidatorFn,
+    Validators
+} from '@angular/forms';
+import {
+    isNull,
+    isNullOrUndefined
+} from 'util';
+import { TerraFormFieldHorizontalContainer } from '../data/terra-form-field-horizontal-container';
+import { TerraFormFieldConditionalContainer } from '../data/terra-form-field-conditional-container';
+
+/**
+ * @author mfrank
+ */
+@Injectable()
+export class TerraFormFieldControlService
+{
+    dynamicFormGroup:FormGroup;
+    formFieldsToGroup:{ [key:string]:FormControl };
+    defaultValues:{ [key:string]:string | number | boolean };
+    translationMapping:{ [key:string]:string };
+
+    constructor(private _formBuilder:FormBuilder)
+    {
+        this.formFieldsToGroup = {};
+        this.defaultValues = {};
+        this.translationMapping = {};
+    }
+
+    private initFormGroupHelper(formFields:Array<TerraFormFieldBase<any>>,
+                               isDisabled:boolean = false):void
+    {
+        formFields.forEach((formField:TerraFormFieldBase<any>) =>
+        {
+            if(formField instanceof TerraFormFieldHorizontalContainer && !isNullOrUndefined(formField.containerEntries))
+            {
+                this.initFormGroupHelper(formField.containerEntries, false);
+            }
+            else if(formField instanceof TerraFormFieldConditionalContainer && !isNullOrUndefined(formField.conditionalEntries))
+            {
+                this.formFieldsToGroup[formField.key] = new FormControl(formField.value, null);
+
+                for(let key in formField.conditionalEntries)
+                {
+                    if(formField.conditionalEntries.hasOwnProperty(key))
+                    {
+                        this.initFormGroupHelper(formField.conditionalEntries[key],  true);
+                    }
+                }
+            }
+            else
+            {
+                this.formFieldsToGroup[formField.key] = new FormControl(formField.value, this.generateValidators(formField));
+                if(isDisabled)
+                {
+                    this.formFieldsToGroup[formField.key].disable({onlySelf: true, emitEvent: false});
+                }
+                this.defaultValues[formField.key] = formField.value;
+                this.translationMapping[formField.key] = formField.label;
```

Comment:
Lässt sich der Block in eine Funktion mit eleganten Namen auslagern, durch welches beim lesen dieses else direkt jeden klar ist was passiert, z.B. `initFormField(formField, isDiabled)`?

Reply:
Definitiv. Code refactoring muss sein :)